packages/frontend-shared/src/components/ResultCounts.vue

```diff
@@ -36,33 +41,50 @@ const { t } = useI18n()
 
 const props = defineProps<ResultCountsProps>()
 
+const ORDER_MAP = {
+  'SKIPPED': {
+    value: props.totalSkipped,
+    class: 'icon-dark-gray-400',
+    icon: SkippedIcon,
+    name: t('runs.results.skipped'),
+  },
+  'PENDING': {
+    value: props.totalPending,
+    class: 'icon-dark-gray-400 icon-light-white',
+    icon: PendingIcon,
+    name: t('runs.results.pending'),
+  },
+  'PASSED': {
+    value: props.totalPassed,
+    class: 'icon-dark-jade-400',
+    icon: PassedIcon,
+    name: t('runs.results.passed'),
+  },
+  'FAILED': {
+    value: props.totalFailed,
+    class: 'icon-dark-red-400',
+    icon: FailedIcon,
+    name: t('runs.results.failed'),
+  },
+}
+
+interface status {
+  value: number|string|null
+  class: string
+  icon: FunctionalComponent<SVGAttributes, {}>
+  name: string
+}
+
```

Comment:
You are off to a good start, but the types can be improved on a bit.

Add a union type for the count types:
`type CountType= 'SKIPPED' | 'PENDING' | 'PASSED' | 'FAILED'`

Then you have the type for `status`, but the convention would be to make the first letter capitalized.  i.e. `Status`

Use the union type for the `order` prop and use it to type the ORDER_MAP
`const ORDER_MAP: Record<CountType, Status>`